bigtable/client/row_range.cc

```diff
@@ -0,0 +1,92 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "bigtable/client/row_range.h"
+
+namespace bigtable {
+inline namespace BIGTABLE_CLIENT_NS {
+namespace btproto = ::google::bigtable::v2;
+
+bool RowRange::IsEmpty() const {
+  std::string unused;
+  // We do not want to copy the strings unnecessarily, so initialize a reference
+  // pointing to *_key_closed() or *_key_open(), as needed.
+  std::reference_wrapper<const std::string> start(unused);
+  bool start_open = false;
+  switch (row_range_.start_key_case()) {
+    case btproto::RowRange::kStartKeyClosed:
+      start = std::cref(row_range_.start_key_closed());
+      break;
+    case btproto::RowRange::kStartKeyOpen:
+      start = std::cref(row_range_.start_key_open());
+      start_open = true;
+      break;
+    case btproto::RowRange::START_KEY_NOT_SET:
+      // A range starting at -infinity is never empty.
```

Comment:
The concept of -infinity for string ranges seems non-standard, since the minimal element for string comparison is the empty string. If we must use it, it should have a comment explaining what we mean.

Another problem is that bigtable.proto says that if start key is not set, it is interpreted as the empty string, inclusive:
https://github.com/googleapis/googleapis/blob/master/google/bigtable/v2/data.proto#L91
so we should do the same here.

Reply:
Indeed, thank you.  I believe I have fixed the code.